reference-implementations/icrc25/signer/src/agent.js

```diff
@@ -23,60 +23,41 @@ export async function call(agent, canisterId, methodName, arg) {
   return await polling.pollForResponse(agent, Principal.from(canisterId), callResponse.requestId, polling.defaultStrategy())
 }
 
-export async function readState(agent, canisterId, requestId, strategy = polling.defaultStrategy()) {
-  return pollForResponse(agent, Principal.from(canisterId), requestId, strategy)
-}
-
-async function pollForResponse(agent, canisterId, requestId, strategy, request) {
-  const path = [new TextEncoder().encode('request_status'), requestId]
-  const currentRequest = request ?? (await agent.createReadStateRequest?.({ paths: [path] }))
-  const state = await agent.readState(canisterId, { paths: [path] }, undefined, currentRequest)
+export async function pollForCert(agent, canisterId, requestId) {
   if (agent.rootKey == null) throw new Error('Agent root key not initialized before polling')
-  const cert = await Certificate.create({
-    certificate: state.certificate,
-    rootKey: agent.rootKey,
-    canisterId: canisterId
-  })
-  const maybeBuf = cert.lookup([...path, new TextEncoder().encode('status')])
-  let status
-  if (typeof maybeBuf === 'undefined') {
-    // Missing requestId means we need to wait
-    status = RequestStatusResponseStatus.Unknown
-  } else {
-    status = new TextDecoder().decode(maybeBuf)
-  }
+  const path = [new TextEncoder().encode('request_status'), requestId]
 
-  // eslint-disable-next-line default-case
-  switch (status) {
-    case RequestStatusResponseStatus.Replied: {
-      return { 
-        response: cert.lookup([...path, 'reply']),
-        certificate: state.certificate
-      }
+  for (;;) {
+    const request = await agent.createReadStateRequest({paths: [path]})
```

Comment:
From the original `pollForResponse` I gather that this could be created just once.